features/FxA-45-device-registration-api/README.md

```diff
@@ -0,0 +1,373 @@
+https://api-accounts.dev.lcip.org/v1/account/device \
+-d '{
+  "id": "3d7d",
+  "name": "My Old Phone"
+}'
+```
+
+#### Response
+
+```
+{
+  "id": "3d7d",
+  "name": "My Phone",
+  "type": "mobile",
+  "callback": "https://updates.push.services.mozilla.com/update/abcdef01234567890abcdefabcdef01234567890abcdef"
+}
+```
+
+### POST /v1/account/device/destroy
+
+Authenticated with session token or OAuth.
+
+#### Request
+
+```
+curl -v \
+-X POST \
+-H "Host: api-accounts.dev.lcip.org" \
+-H "Content-Type: application/json" \
+-H 'Authorization: Hawk id="d4c5b1e3f5791ef83896c27519979b93a45e6d0da34c7509c5632ac35b28b48d", ts="1373391043", nonce="ohQjqb", hash="vBODPWhDhiRWM4tmI9qp+np+3aoqEFzdGuGk0h7bh9w=", mac="LAnpP3P2PXelC6hUoUaHP72nCqY5Iibaa3eeiGBqIIU="' \
+https://api-accounts.dev.lcip.org/v1/account/device/destroy \
+-d '{
+  "id": "3d7d"
+}'
+```
+
+#### Response
+
+```
+{}
+```
+
+### GET /v1/account/devices
+
+Authenticated with session token or OAuth.
+
+#### Request
+
+```
+curl -v \
+-X GET \
+-H "Host: api-accounts.dev.lcip.org" \
+-H "Content-Type: application/json" \
+-H 'Authorization: Hawk id="d4c5b1e3f5791ef83896c27519979b93a45e6d0da34c7509c5632ac35b28b48d", ts="1373391043", nonce="ohQjqb", hash="vBODPWhDhiRWM4tmI9qp+np+3aoqEFzdGuGk0h7bh9w=", mac="LAnpP3P2PXelC6hUoUaHP72nCqY5Iibaa3eeiGBqIIU="' \
+https://api-accounts.dev.lcip.org/v1/account/devices
+```
+
+#### Response
+
+Instead of including the `sessionToken` field,
+each device will have a `status`
+of `"connected"` or `"disconnected"`.
+
+```
+[
+  {
+    "id": "3d7d",
+    "name": "My Phone",
+    "type": "mobile",
+    "status": "connected",
+    "callback": "https://updates.push.services.mozilla.com/update/abcdef01234567890abcdefabcdef01234567890abcdef"
+  },
+  {
+    "id": "01ab",
+    "name": "My Desktop",
+    "type": "desktop",
+    "status": "disconnected",
+    "callback": "https://updates.push.services.mozilla.com/update/d4c5b1e3f5791ef83896c27519979b93a45e6d0da34c75"
+  }
+]
+```
+
+## Details
+
+### What is a "recovering device"?
+
+A recovering device is one that
+has been registered before but has lost its device id.
+We may rely on the device `name` field in this case.
+The recovering device will login as a new device, submitting its name.
+If the name matches an existing device,
+we will consider them to be the same,
+otherwise a new device will be created.
+
+Responding to "We may rely on the device `name` field in this case",
+Ryan K commented:
+
+> I don't think I like this,
+> it seems too easy to have device names that accidentally match,
+> particularly if the device fills in a default based on its hardware platform etc.
+> If we're going to do recovering devices,
+> I think we should let the device provide some sort of fingerprint
+> that it's extremely confident will be unique.
```

Comment:
I think we should explicitly de-scope device recovery from the initial version of this feature, and revisit it in https://github.com/mozilla/fxa-auth-server/issues/1077